### PR TITLE
[Test][Cephadm] Add test to validate pid limit of services

### PR DESCRIPTION
**Pull request by pranavprakash20**
# Description

CEPH-83575590 - RHEL 9 (cgroups v2) - the pid limits ARE enforced as compared to RHEL8 (cgroup v1)

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Comment by pranavprakash20** [comment]
run log: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1ZJM0U/

File: cli/utilities/utils.py [hunk 1]
```diff
@@ -800,3 +800,22 @@ def copy_ceph_sshkey_to_host(installer, node, user="root", key=CEPH_PUB_KEY):
...
+
+def get_pid_limit(node, service):
+    """
+    Gets the pids limit of a given service
+    Args:
+        node (CephNode): Ceph node where the service is running
+        service (str): Name of the service (rgw, mon, mds)
+    """
+    # Identify the process name of the given service
+    cmd = f"podman ps | grep {service}"
```
Comment (vamahaja) [inline cli/utilities/utils.py:813]: Add method for this step here - https://github.com/red-hat-storage/cephci/blob/master/cli/utilities/containers.py#L47

File: cli/utilities/utils.py [hunk 2]
```diff
@@ -800,3 +800,22 @@ def copy_ceph_sshkey_to_host(installer, node, user="root", key=CEPH_PUB_KEY):
...
+    out, _ = node.exec_command(cmd=cmd, sudo=True)
+    process_name = out.split(" ")[0]
+
+    # Check the pids-limit for identified service container
+    cmd = f"podman inspect {process_name}  | grep -i pids-limit"
```
Comment (vamahaja) [inline cli/utilities/utils.py:818]: instead use - https://github.com/red-hat-storage/cephci/blob/master/cli/utilities/containers.py#L151

File: tests/cephadm/test_validate_pid_limit.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+from cli.exceptions import OperationFailedError
+from cli.utilities.utils import get_pid_limit
+from utility.log import Log
+
+log = Log(__name__)
```
Comment (vamahaja) [inline tests/cephadm/test_validate_pid_limit.py:5]: Unused variable

**Review by vamahaja — CHANGES_REQUESTED** [review]

**Comment by pranavprakash20** [comment]
All review comments addressed
New run log: 

http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-64XLB6/

File: cli/utilities/containers.py [hunk 4]
```diff
@@ -148,16 +148,19 @@ def rmi(self, image):
...
         Args:
             image (str): Container image url
             format (str): String to format
+            grep (str): String to grep from output
         """
         cmd = f"{self.base_cmd} inspect {image}"
         if format:
             cmd += f" --format={format}"
+        if grep:
+            cmd += f" | grep -i {grep}"
```
Comment (vamahaja) [inline cli/utilities/containers.py:163]: instead use `--filter` parameter. Refer - https://docs.podman.io/en/stable/markdown/podman-images.1.html

File: cli/utilities/containers.py [hunk 5]
```diff
@@ -179,3 +182,18 @@ def compare(self, image, version):
...
+        Args:
+            all (bool): Show all containers
+            grep (str): String to grep from the output
+        """
+        cmd = f"{self.base_cmd} ps"
+        if all:
+            cmd += " --all"
+
+        if grep:
+            cmd += f" | grep -i {grep}"
```
Comment (vamahaja) [inline cli/utilities/containers.py:198]: same as above, refer - https://docs.podman.io/en/stable/markdown/podman-ps.1.html

**Review by vamahaja — CHANGES_REQUESTED** [review]

**Comment by pranavprakash20** [comment]
new run log: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-9R39ER/

File: cli/utilities/containers.py (hunk 4 above)
Comment (vamahaja) [reply cli/utilities/containers.py:163]: My bad. 

Don't add grep here. Get the data through this lib and then parse it accordingly in the test.

File: cli/utilities/utils.py [hunk 6]
```diff
@@ -800,3 +801,23 @@ def copy_ceph_sshkey_to_host(installer, node, user="root", key=CEPH_PUB_KEY):
...
+    """
+    Gets the pids limit of a given service
+    Args:
+        node (CephNode): Ceph node where the service is running
+        service (str): Name of the service (rgw, mon, mds)
+    """
+    # Identify the process name of the given service
+    out = Container(node).ps(filter={"name": service})[0]
+    process_name = out.split("\n")[1]
+    log.info(process_name)
```
Comment (vamahaja) [inline cli/utilities/utils.py:816]: not much data, please remove

File: cli/utilities/utils.py [hunk 7]
```diff
@@ -800,3 +801,23 @@ def copy_ceph_sshkey_to_host(installer, node, user="root", key=CEPH_PUB_KEY):
...
+    process_name = process_name.split(" ")[0].strip()
+    log.info(process_name)
```
Comment (vamahaja) [inline cli/utilities/utils.py:818]: not much data, please remove

**Review by vamahaja — CHANGES_REQUESTED** [review]

**Review by vamahaja — COMMENTED** [review]
LGTM

**Review by tintumathew10 — APPROVED** [review]
LGTM

**Pull request merged**